### PR TITLE
[Minor] Improve completion of accelerator name when cloud is specified

### PR DESCRIPTION
**Pull request by Michaelvll**
Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/resources.py [hunk 1]
```diff
@@ -501,8 +501,9 @@ def _set_accelerators(
 
             # Canonicalize the accelerator names.
             accelerators = {
-                accelerator_registry.canonicalize_accelerator_name(acc):
-                acc_count for acc, acc_count in accelerators.items()
+                accelerator_registry.canonicalize_accelerator_name(
+                    acc, self._cloud): acc_count
```
Comment (concretevitamin) [inline sky/resources.py:505]: Minor: What happens if a user has 1 cloud only as indicated by `sky check` but `self._cloud` is not specified? I guess we can leave that to the future.

**Review by concretevitamin — APPROVED** [review]
LGTM, thanks @Michaelvll.

File: sky/resources.py (hunk 1 above)
Comment (Michaelvll) [reply sky/resources.py:505]: That is a good point! We can consider inferring the cloud for the whole Resources when only one cloud is enabled. Let's leave it to the future as that may need more test. : )

Comment (Michaelvll) [reply sky/resources.py:505]: Actually, this might be debatable, as this may make an original working yaml fails, if a user add a new cloud.

**Pull request merged**